core/src/main/java/org/apache/servicecomb/core/provider/consumer/InvokerUtils.java

```diff
@@ -35,31 +33,37 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(InvokerUtils.class);
 
   public static Object syncInvoke(String microserviceName, String schemaId, String operationName, Object[] args) {
-    checkEngineStatus();
-    ReferenceConfig referenceConfig = CseContext.getInstance().getConsumerProviderManager()
-        .getReferenceConfig(microserviceName);
+    ReferenceConfig referenceConfig = SCBEngine.getInstance().getReferenceConfigForInvoke(microserviceName);
     return syncInvoke(generateInvocation(schemaId, operationName, args, referenceConfig));
   }
 
   public static Object syncInvoke(String microserviceName, String microserviceVersion, String transport,
       String schemaId, String operationName, Object[] args) {
-    checkEngineStatus();
-    ReferenceConfig referenceConfig = CseContext.getInstance().getConsumerProviderManager()
-        .createReferenceConfig(microserviceName, microserviceVersion, transport);
+    ReferenceConfig referenceConfig = SCBEngine.getInstance()
+        .createReferenceConfigForInvoke(microserviceName, microserviceVersion, transport);
     return syncInvoke(generateInvocation(schemaId, operationName, args, referenceConfig));
   }
 
+  /**
+   * it's a internal API, caller make sure already invoked SCBEngine.ensureStatusUp
+   * @param invocation
+   * @return
+   * @throws InvocationException
+   */
   public static Object syncInvoke(Invocation invocation) throws InvocationException {
-    checkEngineStatus();
     Response response = innerSyncInvoke(invocation);
     if (response.isSuccessed()) {
       return response.getResult();
     }
     throw ExceptionFactory.convertConsumerException(response.getResult());
   }
 
+  /**
+   * it's a internal API, caller make sure already invoked SCBEngine.ensureStatusUp
+   * @param invocation
+   * @return
```

Comment:
It's better to specify the return response.

Reply:
done